extension/script.js

```diff
@@ -14,16 +20,64 @@ const {
 //})
 
 if (typeof window.web3 !== 'undefined') {
-  throw new Error('web3 already exists.')
+  //throw new Error('web3 already exists.')
+  console.error("Gnosis Safe overrode an existing web3, please disable the whitelisting or uninstall one to prevent this.")
 }
 
 var engine = new ProviderEngine()
+engine.isMetaMask = !0
+engine.isConnected = function() { return true; }
 const gnosisProvider = new GnosisProvider()
 engine.addProvider(gnosisProvider)
-engine.addProvider(new RpcSubprovider({
+
+// static
+const staticSubprovider = new DefaultFixture()
+engine.addProvider(staticSubprovider)
+
+// nonce tracker
+engine.addProvider(new NonceTrackerSubprovider())
+
+// sanitization
+const sanitizer = new SanitizingSubprovider()
+engine.addProvider(sanitizer)
+
+// cache layer
+const cacheSubprovider = new CacheSubprovider()
+engine.addProvider(cacheSubprovider)
+
+const filterAndSubsSubprovider = new SubscriptionSubprovider()
+// forward subscription events through provider
+filterAndSubsSubprovider.on('data', (err, notification) => {
+  engine.emit('data', err, notification)
+})
+engine.addProvider(filterAndSubsSubprovider)
+
+// inflight cache
+const inflightCache = new InflightCacheSubprovider()
+engine.addProvider(inflightCache)
+
+engine.addProvider(new FetchSubprovider({
   rpcUrl: config.networks[config.currentNetwork].url,
 }))
 
+engine.send = function(payload){
```

Comment:
It would probably be better to subclass the ProviderEngine and do everything that I did in this class there.